ztoc/ztoc_marshaler.go

```diff
@@ -238,3 +256,16 @@ func prepareXattrsOffset(me FileMetadata, builder *flatbuffers.Builder) flatbuff
 	xattrs := builder.EndVector(len(me.Xattrs))
 	return xattrs
 }
+
+// compressionAlgorithmToFlatbuf helps convert compression algorithm into flatbuf
+// enum. SOCI/containerd uses lower-case for compression, but our flatbuf capitalizes
+// the first letter. When converting back, we can just `strings.ToLower` so a helper
+// func is not needed in that case.
+func compressionAlgorithmToFlatbuf(algo string) (ztoc_flatbuffers.CompressionAlgorithm, error) {
```

Comment:
If we stored `algo` as a byte, wouldn't the flatbuffer serialization logic handle this for us? i.e. if you tried to write a value that wasn't present, then it would cause a serialization error.

Reply:
I'm not sure how will that work (also by `byte`, do you mean something like `[]byte("gzip")`?). [flatbuf generates its enum as int](https://github.com/awslabs/soci-snapshotter/blob/main/ztoc/fbs/ztoc/CompressionAlgorithm.go).

Regardless, I feel a conversion is unavoidable and it's more convenient to keep it consistent with containerd (which returns "gzip"/"zstd"). If we use flatbuffer's byte/int, we still need to convert that to string or vice versa when we need to compare.

Reply:
We discussed offline and decided that this is convenient for now. In the future, we may decide that compression algorithms are actually died to the specific implementation of the algorithm (e.g. a C implementation and a go implementation would have different `CompressionAlgorithm` fields in the ztoc) at which point we can consider representing this differently.